core/variable_map.js

```diff
@@ -81,15 +81,21 @@ Blockly.VariableMap.prototype.renameVariable = function(variable, newName) {
   }
 
   if (variableIndex == -1 && newVariableIndex == -1) {
-    this.createVariable(newName, '');
+    var newVar = this.createVariable(newName, '');
     console.log('Tried to rename an non-existent variable.');
   } else if (variableIndex == newVariableIndex ||
       variableIndex != -1 && newVariableIndex == -1) {
     // Only changing case, or renaming to a completely novel name.
-    this.variableMap_[type][variableIndex].name = newName;
+    var renameVar = this.variableMap_[type][variableIndex];
```

Comment:
I'd prefer `variableToRename` and `variableToDelete`, to make it clearer that you are
1) finding the variable
2) firing an event
3) taking the named action on that variable